pkg/download/yamlcreator/yaml_creator.go

```diff
@@ -26,24 +27,53 @@ import (
 
 //YamlCreator implements method to create the yaml configuration file
 type YamlCreator interface {
+	ReadYamlFile(fs afero.Fs, configSubPath string, apiId string) error
 	CreateYamlFile(fs afero.Fs, path string, name string) error
```

Comment:
nitpick: Now that you've introduced reading/updating I'd rename this method to `WriteYamlFile` to fit what it's really doing - creating or overwriting